cgi/user.pl

```diff
@@ -118,6 +118,17 @@
 SCRIPT
 ;
 
+	my $user_info = remove_tags_and_quote(param('user_id'));
+	$user_info =~ /^(.+?)@/;
+	if ( defined ($1) ){
+		$user_ref->{email} = $user_info;
+		$user_ref->{userid} = $1;
+		$user_ref->{name} = $1;
+	}
+	else{
+		$user_ref->{userid} = $user_info;
```

Comment:
Now after reading the code that handles it in `User.pm` I feel that we should not set the `userid` at all.
What do you say ?

Reply:
for keeping the `code` do you think I should make another CGI object which I can reference in `Users.pm` to create the similar effect as the line `<input type="hidden" name="code" value="$code" />`does in `product_multilingual.pl` ?

Reply:
We could do the following if you enter a username + a password and click on the create button:

1. in the session / create user form in product_multilingual.pm : add a hidden field so that we know we are coming from this form. (e.g. <input type="hidden" name="from" value="login_or_register" />

2. in user.pl , before we call init() (which calls the functions in Users.pm to check if the username and password exist),  look for this hidden field. If we have it, record the value of the password parameter, and delete the password parameter (Something like param("password", "") should do it). --> that will prevent the password check after.

3. use the recorded password to initialize the value.

Reply:
Updated, now it works with password field too.
How do I make it redirect to the `product_code` edit page as @hangy suggested ?
Thanks

Reply:
When editing, `product.pl`  is called with the `code` URL parameter, which is passed to `user.pl` in https://github.com/openfoodfacts/openfoodfacts-server/blob/80312221191ba62f956526c2ddb5a01fa8c45b78/cgi/product_multilingual.pl#L226 An easy option could be to add this to a hidden `input`  field in `user.pl`, check if it's present after the user registered, and redirect them to `product.pl`.

Reply:
@muskangarg21 : when we open a session, there is already a redirect to the product edit page, it's made with this code in session.pl:

You could so Something similar in user.pl when there is the next_action parameter.

````
	my $next_action = param('next_action');
	my $code = param('code');
	my $r = shift;
	my $referer = $r->headers_in->{Referer};
	my $url;

	if ((defined $next_action) and ($code =~ /^(\d+)$/)) {
		if ($next_action eq 'product_add') {
			$url = "/cgi/product.pl?type=add&code=$code";
		}
		elsif ($next_action eq 'product_edit') {
			$url = "/cgi/product.pl?type=edit&code=$code";
		}
	}
	elsif ((defined $referer) and ($referer =~ /^https?:\/\/$subdomain\.$server_domain/) and (not ($referer =~ /(?:session|user)\.pl/))) {
		$url = $referer;
	}

	if (defined $url) {

		$log->info("redirecting after login", { url => $url }) if $log->is_info();

        $r->err_headers_out->add('Set-Cookie' => $cookie);
		$r->headers_out->set(Location =>"$url");
		$r->status(301);
		return 301;
	}

Reply:
@stephanegigandet @hangy 
I tried qutie a few things but the major problem is that in `Sign in` button the form action itself is the session.pl where as in `Create account` the form action is `user.pl` which displays a form whose `Submit Query` has another formaction linked to it, now the problem I am facing is that whatever I do in `user.pl` gets executed when we press the `Create Account` button so I am not able to get whether the user has entered valid details or not when hitting the `Submit Query` button.
Little help needed.
Thanks